### ISSUE TITLE
Bug report for calendar tree view

### ISSUE THREAD

**Issue by dominiquekonsuntack** [odoo/odoo#26908]
Impacted versions:
10 - tested on runbot --> show the error
11 is not affected

Steps to reproduce:
- Go to calendar, create a calendar event, without any attendees e.g. Christmas celebration
- Then create another calendar event with an attendee
- Switch to list view, all the events are visible, as well as the attendees
- Then click on the column "Attendee" to sort by attendee

Current behavior:
```
Error message:
   response = f(*args, **kw)
  File "/home/odoo/runbot/runbot/runbot/static/build/371321-10-0-077630/odoo/addons/web/controllers/main.py", line 837, in search_read
    return self.do_search_read(model, fields, offset, limit, domain, sort)
  File "/home/odoo/runbot/runbot/runbot/static/build/371321-10-0-077630/odoo/addons/web/controllers/main.py", line 859, in do_search_read
    offset=offset or 0, limit=limit or False, order=sort or False)
  File "/home/odoo/runbot/runbot/runbot/static/build/371321-10-0-077630/odoo/models.py", line 4704, in search_read
    records = self.search(domain or [], offset=offset, limit=limit, order=order)
  File "/home/odoo/runbot/runbot/runbot/static/build/371321-10-0-077630/odoo/addons/calendar/models/calendar.py", line 1599, in search
    events = self.browse(events.get_recurrent_ids(args, order=order))
  File "/home/odoo/runbot/runbot/runbot/static/build/371321-10-0-077630/odoo/addons/calendar/models/calendar.py", line 1100, in get_recurrent_ids
    ids = [r['id'] for r in sorted(result_data, cmp=comparer)]
  File "/home/odoo/runbot/runbot/runbot/static/build/371321-10-0-077630/odoo/addons/calendar/models/calendar.py", line 1091, in comparer
    result = cmp(fn(left), fn(right))
  File "/home/odoo/runbot/runbot/runbot/static/build/371321-10-0-077630/odoo/models.py", line 5197, in __gt__
    raise TypeError("Mixing apples and oranges: %s > %s" % (self, other))
TypeError: Mixing apples and oranges: res.partner() > YourCompany, Administrator
```
Expected behavior:

in V11, it is working fine, the event with no attendee goes to the top of the list without any error

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
@Yenthe666 the bug is in v10, and I think seeing the stack-trace, that is more something of framework.

**Comment by sswapnesh** [comment]
@pedrobaeza I believe it is from `calendar` module itself.
https://github.com/odoo/odoo/blob/10.0/addons/calendar/models/calendar.py#L1092-#L1103

**Comment by Yenthe666** [comment]
Yup, my bad on the versioning, sorry!

**Comment by pedrobaeza** [comment]
@sswapnesh yeah, the problem can be in that part. @mart-e can you take a look to this?

**Comment by mart-e** [comment]
Thanks for the report, fixed at 6dec0bc308616fb4c5818470c7fed112bde2517c
